Tests/TestUtils/TestBase.swift

```diff
@@ -223,4 +223,25 @@ class TestBase: XCTestCase {
         guard !message.isEmpty && TestBase.debugEnabled else { return }
         print("TestBase - \(message)")
     }
+
+    // MARK: Specialized Helpers
+    func getEdgeEventHandles(expectedHandleType: String) -> [Event] {
+        return getDispatchedEventsWith(type: TestConstants.EventType.EDGE, source: expectedHandleType)
+    }
+
+    func getEdgeResponseErrors() -> [Event] {
+        return getDispatchedEventsWith(type: TestConstants.EventType.EDGE, source: TestConstants.EventSource.ERROR_RESPONSE_CONTENT)
+    }
+
+    /// Extracts the Edge location hint from the location hint result
+    func getLastLocationHintResultValue() -> String? {
+        let locationHintResultEvent = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT).last
+        guard let payload = locationHintResultEvent?.data?["payload"] as? [[String: Any]] else {
+            return nil
+        }
+        guard payload.indices.contains(2) else {
+            return nil
+        }
+        return payload[2]["hint"] as? String
+    }
```

Comment:
Since testBase is for Edge Network we could keep those edge specific still basic to be used across tests. I don't think it is a good idea to pass around testBase to helpers. 

If we decide to put testBase to a separate package, I think a better Idea would be to create a new helper which will extend the testBase class and add those helpers. 

```
// filename: TestBase+Helpers

extension TestBase {
 func getLastLocationHintResultValue() -> String? {
        let locationHintResultEvent = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT).last
        guard let payload = locationHintResultEvent?.data?["payload"] as? [[String: Any]] else {
            return nil
        }
        guard payload.indices.contains(2) else {
            return nil
        }
        return payload[2]["hint"] as? String
    }

}